cpp/src/tests/sssp/sssp_test.cu

```diff
@@ -0,0 +1,419 @@
+        std::cout << SSSP_time[i] / PERF_MULTIPLIER << std::endl;
+      }
+    }
+  }
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+
+  static std::vector<double> SSSP_time;
+
+  template <typename DistType, bool DoDist, bool DoPreds>
+  void run_current_test(const SSSP_Usecase& param) {
+    gdf_column col_src, col_dest, col_weights, col_distances, col_preds;
+
+    size_t v, e;
+    // Input
+    col_src.data = nullptr;
+    col_src.dtype = GDF_INT32;
+    col_src.valid = nullptr;
+    col_src.null_count = 0;
+
+    col_dest.data = nullptr;
+    col_dest.dtype = GDF_INT32;
+    col_dest.valid = nullptr;
+    col_dest.null_count = 0;
+
+    col_weights.data = nullptr;
+    if (std::is_same<DistType, float>::value)
+      col_weights.dtype = GDF_FLOAT32;
+    else if (std::is_same<DistType, double>::value)
+      col_weights.dtype = GDF_FLOAT64;
+    else
+      ASSERT_TRUE(0);
+    col_weights.valid = nullptr;
+    col_weights.null_count = 0;
+
+    // Output
+    col_distances.data = nullptr;
+    col_distances.size = 0;
+    col_preds.data = nullptr;
+    col_preds.size = 0;
+
+    if (param.type_ == RMAT) {
+      ASSERT_EQ(
+          gdf_grmat_gen(
+              param.config_.c_str(), v, e, &col_src, &col_dest, &col_weights),
+          GDF_SUCCESS);
+    } else if (param.type_ == MTX) {
+      int m, k, nnz;
+      MM_typecode mc;
+
+      FILE* fpin = fopen(param.file_path_.c_str(), "r");
+
+      ASSERT_EQ(mm_properties<int>(fpin, 1, &mc, &m, &k, &nnz), 0)
+          << "could not read Matrix Market file properties"
+          << "\n";
+      ASSERT_TRUE(mm_is_matrix(mc));
+      ASSERT_TRUE(mm_is_coordinate(mc));
+      ASSERT_FALSE(mm_is_complex(mc));
+      ASSERT_FALSE(mm_is_skew(mc));
+
+      // Allocate memory on host
+      std::vector<int> cooRowInd(nnz), cooColInd(nnz);
+      std::vector<DistType> cooVal;
+
+      // Read weights if given
+      if (!mm_is_pattern(mc)) {
+        cooVal.resize(nnz);
+        ASSERT_EQ(
+            (mm_to_coo<int, DistType>(
+                fpin, 1, nnz, &cooRowInd[0], &cooColInd[0], &cooVal[0], NULL)),
+            0)
+            << "could not read matrix data"
+            << "\n";
+      } else {
+        ASSERT_EQ((mm_to_coo<int, DistType>(
+                      fpin, 1, nnz, &cooRowInd[0], &cooColInd[0], NULL, NULL)),
+                  0)
+            << "could not read matrix data"
+            << "\n";
+        // Set random weights
+        if (std::is_same<DistType, float>::value ||
+            std::is_same<DistType, double>::value) {
+          cooVal.resize(nnz);
+          for (auto i = 0; i < nnz; i++) {
+            cooVal[i] = static_cast<DistType>(rand()) /
+                static_cast<DistType>(RAND_MAX);
+          }
+        }
+      }
+
+      ASSERT_EQ(fclose(fpin), 0);
+      // gdf columns
+      create_gdf_column(cooRowInd, &col_src);
+      create_gdf_column(cooColInd, &col_dest);
+      create_gdf_column(cooVal, &col_weights);
```

Comment:
Where are you releasing memory allocated in the above three lines?

I know it is very confusing, but in gdf_edge_list_view(), ownership is set to 0, the resulting graph does not release memory pointed by the gdf_column's data. So you need to manually release memory once the graph object is destroyed and you're done. Also, note that there is another version of create_gdf_column() which returns std::unique_ptr and releases memory when the std::unique_ptr object becomes out of scope (in this case, you need to make sure that this object out-lives the graph object).

Reply:
Ack. Added free in revised commit.